ledger/src/blockstore.rs

```diff
@@ -3680,6 +3685,8 @@ fn update_slot_meta(
     // set it to this shred index
     if is_last_in_slot && slot_meta.last_index.is_none() {
         slot_meta.last_index = Some(u64::from(index));
+        let slot_time_elapsed = u64::from(reference_tick) * 1000 / DEFAULT_TICKS_PER_SECOND;
+        slot_meta.last_shred_timestamp = timestamp() - slot_time_elapsed;
```

Comment:
I don't think you would need to adjust. `last_shred_timestamp` would be adjusted to be the prediction of the first shred here.